modules/search-service-impl/src/main/java/org/opencastproject/search/endpoint/SearchRestService.java

```diff
@@ -223,6 +223,47 @@ public Response remove(@PathParam("id") String mediaPackageId) throws SearchExce
     }
   }
 
+  @DELETE
+  @Path("removeSynchronously/{id}")
+  @Produces(MediaType.APPLICATION_XML)
+  @RestQuery(
+      name = "remove",
+      description = "Removes a mediapackage from the search index.",
+      pathParameters = {
+          @RestParameter(
+              description = "The media package ID to remove from the search index.",
+              isRequired = true,
+              name = "id",
+              type = RestParameter.Type.STRING
+          )
+      },
+      responses = {
+          @RestResponse(description = "OK", responseCode = HttpServletResponse.SC_OK),
+          @RestResponse(
+              description = "The mediapackage could not be delete by you or does not exist",
+              responseCode = HttpServletResponse.SC_BAD_REQUEST
+          ),
+          @RestResponse(
+              description = "There has been an internal error and the mediapackage could not be deleted",
+              responseCode = HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+          )
+      },
+      returnDescription = "Nothing"
+  )
+  public Response removeSynchronously(@PathParam("id") String mediaPackageId) throws SearchException {
+    try {
+      boolean removed = searchService.deleteSynchronously(mediaPackageId);
+      if (removed) {
+        return Response.ok().build();
+      } else {
+        return Response.status(Response.Status.BAD_REQUEST).build();
+      }
+    } catch (Exception e) {
+      logger.info("Unable to remove mediapackage {} from search index: {}", mediaPackageId, e.getMessage());
```

Comment:
Is this an actual error? In that case, do we want this to be logged as an error including a stack trace? Or is this something that appears if you request a non-existing event and this should be more like a debug message?

Reply:
I'm not exactly sure what you are asking. This catches everything that can go wrong in `deleteSynchronously`. I can change the log call to include the stack trace, sure. This pattern is mostly copied from all the other "wrappers" in this file. 🤔🤷‍♀️

Reply:
He (and I) would be temped to handle things that are Exceptions-but-not-errors like deleting a non-existent mediapackage differently. By all means have a catch all, but don't log a huge stack if the reason is user error :D

Reply:
Deleting a non-existent MP should make `deleteSynchronously` return `false`, so that's handled.

As far as I can see this should only happen in truly exceptional cases. In which case we might want to log the stacktrace, fair, but that's not done anywhere else in this file.

I can change it if you (as in any of you) insist, but I also find it a bit weird to only change it in this method (and maybe also just in this file? haven't looked at other remote impls, though), in which case it's beyond the scope of this PR IMO.

Reply:
The e.getMessage() is not the stack trace. It might be helpful to add an accompanying TRACE log to log the actual stack trace, "e", in case that INFO log indicates something difficult to debug.

Is there an INFO log somewhere else that logs that this mediapackage was requested to be deleted (and the user, if known)?

Reply:
I don't think this remote impl should log anything that doesn't have to do with its job of forwarding requests to the actual impl. If we want a log about users deleting things from search, this is not the place IMO. And again, the existing methods in here don't do that, either. 🤷‍♀️